flamedisx/inference.py

```diff
@@ -391,12 +396,69 @@ def parse_result(self, result: Minuit):
         position = {k: result.fitarg[k] for k in self.arg_names}
         return position, result.fval
 
+class NonlinearObjective(Objective):
+    """Compute limits on target parameter given nonlinear constraint
+    defined by likelihood ratio.
+    """
+
+    require_complete_guess = False
+
+    # TODO: Fix callback
+    # Callback function to capture intermediate states of optimizer
+    #def fishPath(self, a, b):
+    #    self._callbackbag.append(a)
+
+    def _inner_fun_and_grad(self, params):
+        # lol who exactly is this Objective._inner_fun_and_grad(self, params)?
+        m2ll, grad, hess = Objective._inner_fun_and_grad(self, params)
+        return m2ll - self.m2ll_best - self.t_ppf(params), grad - self.t_ppf_grad(params), hess - self.t_ppf_hess(params)
+
+    def tilt_fun(self, x):
+        return -1.*x
+
+    def hess_constraint(self, x, v):
+        # TODO: Must return hessian matrix of dot(fun, v), is that what this
+        # does?
+        return np.dot(self(x).hess, v)
```

Comment:
Currently tests are failing here:

```python
return np.dot(self(x).hess, v)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (array([[1.87668812e+05, 1.82008243e-05],
       [1.82017684e-05, 2.03541267e-10]]), array([0.]))
kwargs = {}
relevant_args = (array([[1.87668812e+05, 1.82008243e-05],
       [1.82017684e-05, 2.03541267e-10]]), array([0.]), None)

>   ???
E   ValueError: shapes (2,2) and (1,) not aligned: 2 (dim 1) != 1 (dim 0)
```

Reply:
Looks like this was resolved (0677d4722320fc89d3445322cbd060c77ed5f818 ?) since the tests are passing